src/main/java/com/google/devtools/build/lib/bazel/bzlmod/BazelLockFileModule.java

```diff
@@ -210,10 +225,15 @@ private boolean shouldKeepExtension(
     // that irrelevant changes (e.g. locations or imports) don't cause the extension to be removed.
     // Note: Extension results can still be stale for other reasons, e.g. because their transitive
     // bzl hash changed, but such changes will be detected in SingleExtensionEvalFunction.
+    var usagesValue = SingleExtensionUsagesFunction.createValue(depGraphValue, extensionId);
+    if (usagesValue.isEmpty()) {
+      // The extension is no longer used anywhere, drop it.
+      return false;
+    }
     return Arrays.equals(
-        ModuleExtensionUsage.hashForEvaluation(
-            GsonTypeAdapterUtil.createModuleExtensionUsagesHashGson(),
-            moduleResolutionEvent.getExtensionUsagesById().row(extensionId)),
+        SingleExtensionUsagesValue.hashForEvaluation(
```

Comment:
this part is now getting quite annoying. I wonder if it's okay to just skip this and fall back to the comment on line 226 and say "well, the extension could be stale for multiple reasons anyway, so let's not bother checking if the usages hash is up to date". WDYT?

Reply:
It did become unnecessary when we introduced the usages hash, but I didn't notice that back then. I agree it's okay to get rid of it.